machine.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package gomaasapi
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/schema"
+	"github.com/juju/version"
+)
+
+type machine struct {
+	// Add the controller in when we need to do things with the machine.
+	// controller Controller
+	resourceURI string
+
+	systemID    string
+	hostname    string
+	fqdn        string
+	ipAddresses []string
+	memory      int
+	cpuCount    int
+	powerState  string
+
+	zone *zone
+
+	operatingSystem string
+	distroSeries    string
+	architecture    string
+	// NOTE: status is an int, need to get status values from MAAS
+	// for now we'll set status to be the status_name
+	status string
```

Comment:
Shouldn't it be statusName then? In juju we only use status_name and not the status int anyway, so I'm not very bothered. To other users of MAAS/gomaasapi it might be confusing though.